subprojects/core/src/main/java/org/gradle/api/internal/file/copy/DefaultCopySpec.java

```diff
@@ -208,20 +216,24 @@ public void addChildSpecListener(CopySpecListener copySpecListener) {
         this.listeners.add(copySpecListener);
     }
 
+    @VisibleForTesting
```

Comment:
Public visibility for testing seems wrong...

Reply:
Having unit tests accessing this from other packages seems wrong, but at least now it's marked for testing. It's not making things worse.